app/course/services/courseStateService.js

```diff
@@ -263,7 +263,12 @@ courseApp.service('courseStateService', function ($rootScope, $log, Course, Term
 					sectionGroups.list[action.payload.sectionGroup.id].plannedSeats = action.payload.sectionGroup.plannedSeats;
 					return sectionGroups;
 				case FETCH_SECTIONS:
-					sectionGroups.list[action.payload.sectionGroup.id].sectionIds = action.payload.sections
+					var sectionGroup = sectionGroups.list[action.payload.sectionGroup.id];
```

Comment:
Please add a comment as to why the best course of action for FETCH_SECTIONS when a sectionGroup is missing is to leave the sectionGroups alone. (Should this be throwing a console warning or reportJS?)

Reply:
Oh, I meant to add a little more detail on this one. This bug is a little unusual.
First a query is run for sections via a sectionGroupId, and that sectionGroup exists in the state.

When the query completes, it attempts to add those sections to the specified sectionGroup, and at this point (on this one occasion) the sectionGroup was no longer in the state.

The only way I can think of to cause this would be if someone on slow internet were to:
1) Click a cell, which triggers multiple api calls including the section fetch
2) Immediately delete the sectionGroup or course row containing that cell
3) Deletion could complete first, removing the sectionGroup from the state before the section fetching completed.

To properly handle slower internet we could:
1) Lock the UI down until the query completes. I see apps do this when its critical to always show the correct thing (like banking apps).
2) Since the sections now reference a sectionGroup that doesn't exist, the data just doesn't matter anymore, in which case, there isn't really a problem, and we should just do nothing.

Reducers are supposed to be as simple as possible and fine with rapid changes in new variations, so keeping the reducer un-opinionated makes sense to me.

I could throw an error explicitly in the action creator to try to identify other scenarios where this might happen, though it typically wouldn't have access to the state so I would need to pass it that. Also, the only additional information that would be helpful here would be a dump of the state history, with specific actions for 'initiating' and 'completing' api calls. That would be pretty awesome actually.

Alternatively, we could leave the code as is and wait and see if it happens again.